packages/ui-components/src/components/BreadcrumbItem/BreadcrumbItem.test.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as React from "react"
+import { render, screen } from "@testing-library/react"
+import { describe, expect, test, vi } from "vitest"
+
+import { Button } from "../Button"
+import { BreadcrumbItem } from "."
+
+describe("BreadcrumbItem", () => {
+  describe("Default Rendering", () => {
+    test("renders without any props", () => {
+      render(<BreadcrumbItem />)
+      expect(screen.getByText("Item")).toBeInTheDocument()
+    })
+
+    test("renders default icon when 'icon' prop is not passed", () => {
+      render(<BreadcrumbItem />)
+      const svgElement = screen.getByRole("img")
+      expect(svgElement).toBeInTheDocument()
+      const titleElement = screen.getByText("Help", { selector: "title" })
+      expect(titleElement).toBeInTheDocument()
+    })
+  })
+
+  describe("Handling Props", () => {
+    test("renders a breadcrumb item with text as passed", () => {
+      render(<BreadcrumbItem data-testid="breadcrumbitem" />)
+      expect(screen.getByTestId("breadcrumbitem")).toBeInTheDocument()
+      expect(screen.getByTestId("breadcrumbitem")).toHaveClass("juno-breadcrumb-item")
+    })
+
+    test("renders an icon as passed", () => {
```

Comment:
Test was failing. I think the structure changed in `Icon` during migration to `Typescript`. 
Please double check that how I updated the test is what's expected. 
Otherwise, there might be a bug in `Icon`.

Reply:
This test probably belongs in `Icon` and not `BreadcrumbItem` too...

Reply:
what exactly failing here?

Reply:
CC: @edda

Reply:
hmm, regarding whether this test should be here: yes, it should. It's not about testing that the image has the correct title attribute, that is just the way that was chosen here to verify that not only is an icon rendered, it's also the correct one. You're right, it's more of an integration test but if it succeeds then it verifies also, that nothing weird is happening within the BreadcrumbItem component.

Regarding the failing test, this is unexpected. The Icon component hasn't changed structurally with the migration. My guess is that the assertion `toHaveAttribute` doesn't work the same in vitest as it did in jest. The way `Icon` works is that we build a component for each svg and add a bunch of props to the component. One of the props is `title`. However the rendered svg doesn't get a title attribute, instead a `<title>` tag is added inside of the svg. In jest the `toHaveAttribute("title", ..)` assertion seems to have correctly evaluated the title tag inside the svg and I'm guessing in vitest it doesn't. Could you try to verify that that is the problem? In that case we could instead of the title assert on the `alt` attribute.

Reply:
Indeed, good point that it could be jest vs. vite. I believe so. In the updated test, I'm querying for the `title` element inside the image/svg (instead of attribute), is that not enough?🙈

Reply:
My main concern regarding this is that it assumes the icon to be an svg. If that ever changes then this test would break. That's why I suggested asserting on the `alt` attribute instead since that will be present irrespective of image type.

Reply:
Ah, got it. Maybe the naming is the problem. It's just searching for an `img`, I think? Not necessarily an svg. But I update the test to use `alt`, more straight forward.

Reply:
No, it searches for an element that has the aria role `img`. This encompasses everything that's an image, irrespective of type.